src/vmemcache.c

```diff
@@ -102,17 +105,17 @@ vmemcache_newU(const char *dir, size_t max_size, size_t fragment_size,
 			goto error_free_cache;
 		}
 
+		if ((size_t)size < cache->size) {
```

Comment:
> So if size of DDAX is not a multiple of Mmap_align you won't allow it. Is this really what you want?

That's not true. I won't allow if the size of a DDAX is smaller than the required minimum size of the cache given by the `max_size` argument (rounded up to multiple of Mmap_align).

> Shouldn't we require size to be determined by the library in case of DDAX (just like we do in pmdk)?

The library checks the size of DDAX and verifies if it is large enough to fit the cache of required size (`max_size`). I do it like in PMDK: https://github.com/pmem/pmdk/blob/master/src/libvmem/vmem.c#L179
and even better, because there is probably a bug (if the size of DDAX is smaller then required size).

> BTW, we really need documentation! Such change should be accompanied by a change to documentation.

That's right. It will be added soon, but not in this PR. This is just a bug fix.

Reply:
For consistency sake, I think we should just map the entire device if the max_size == 0 and the file is devdax, just like in libpmemobj. If `max_size` is != 0 or  `max_size` != actual_size for device dax, then that's an error.

libvmem's behavior is odd - we map the entire thing anyway, and then artificially limit the heap size. This is what needs fixing ;)

Reply:
Consider this scenario:
max_size = 8MB-4K
cache->size = roundup(max_size, 2MB) = 8MB
size = util_file_get_size(devdax) = 8MB-4K

size < cache->size is true, so opening fails

Reply:
@marcinslusarz Indeed, you are right, I was wrong :-)

Reply:
Fixed